test/test_game.py

```diff
@@ -280,9 +290,96 @@ def test_penalization_after_invalid_moves(
 
     def test_board_game_is_class(self):
-        game = WumpusGame()
+        game = patched_game()
         self.assertIsInstance(game._board, Board)
 
+    @parameterized.expand([
+        ('shoot opponent', 5, 8, 8, WEST, 8, 7, 16100, 4, "     "),
+        ('shoot empty cell', 5, 8, 8, EAST, 8, 9, 1100, 4, "##F##"),
+        ('shoot hole', 5, 8, 8, NORTH, 7, 8, 1100, 4, "  O  "),
+        ('shoot own char', 5, 8, 8, SOUTH, 9, 8, 900, 4, "  B  "),
+    ])
+    def test_execute_action_shoot(self, name,
+                                  initial_arrows,
+                                  from_row, from_col, direction,
+                                  destiny_row, destiny_col,
+                                  expected_score, expected_arrows, expected_destiny_cell):
+        game = patched_game()
+
+        game.player_1 = SHOOTER_PLAYER
+        game.current_player = game.player_1
+
+        game._board._board = deepcopy(SCENARIOS_SHOOT_TEST)
+        game.current_player.score = 1000
+        game.current_player.arrows = initial_arrows
+
+        game.execute_action(SHOOT, from_row, from_col, direction)
+
+        current_score = game.current_player.score
+        current_arrows = game.current_player.arrows
+        destiny_cell = game._board._board[destiny_row][destiny_col]
+
+        self.assertEqual(current_score, expected_score)
+        self.assertEqual(current_arrows, expected_arrows)
+        self.assertEqual(destiny_cell.to_str(PLAYER_1), expected_destiny_cell)
+
+    @parameterized.expand([
+        ('cell with hole', 4, 4, WEST, 4, 3, 1100, 2, EMPTY_CELL, '  O  '),
+        ('cell empty', 4, 4, EAST, 4, 5, 1100, 3, EMPTY_CELL, '  B  '),
+        ('cell with opponent char', 4, 4, NORTH, 3, 4, 1100, 2, EMPTY_CELL, '  P  '),
+        ('cell with own char', 4, 4, SOUTH, 5, 4, 900, 3, '  B  ', '  B  '),
+    ])
+    def test_execute_action_move(self, name,
+                                 from_row, from_col, direction,
+                                 destiny_row, destiny_col,
+                                 expected_score, expected_remaining_characters,
+                                 expected_initial_cell, expected_destiny_cell, ):
+        game = patched_game()
+        board, player_1, player_2 = generate_board_from_move_actio()
+        game.player_1 = player_1
+        game.player_2 = player_2
+        game.current_player = game.player_1
+        game._board._board = board
+        game.current_player.score = 1000
+
+        game.execute_action(MOVE, from_row, from_col, direction)
+
+        current_score = game.current_player.score
+        initial_cell = game._board._board[from_row][from_col]
+        destiny_cell = game._board._board[destiny_row][destiny_col]
+        current_remaining_characters = len(game.current_player.characters)
+
+        self.assertEqual(current_score, expected_score)
+        self.assertEqual(initial_cell.to_str(PLAYER_1), expected_initial_cell)
```

Comment:
Why is this being asserted? Initial cell should be known by the test as it is a param of it.

Reply:
You can add logs to tests that are only printed when you execute a test in a verbose way. If this is only to debug, then it should be changed for a log